webapp/src/app/components/dialogs/load-map/load-map.component.ts

```diff
@@ -67,13 +73,31 @@ export class LoadMapComponent {
 		this.mapsSource.data = this.virtualRoot.children || [];
 		this.ref.detectChanges();
 	}
+
+	private async showConfirmDialog() {
+		const hasUnsavedChanges = await firstValueFrom(this.eventsService.hasUnsavedChanges);
```

Comment:
you used `this.eventsService.hasUnsavedChanges.getValue()` in previous code. Is there a reason why it's different here?

Reply:
onUnload cannot be async so I had to use `getValue` which I consider bad practice